tests/Fixer/Operator/TernaryToNullCoalescingFixerTest.php

```diff
@@ -174,6 +176,18 @@ public static function provideFixCases(): iterable
 # c7
 ;',
             ],
+            [
+                '<?php $x = $THIS ?? null;',
+                '<?php $x = isset($THIS) ? $THIS : null;',
+            ],
+            [
+                '<?php $x = $this->foo ?? null;',
+                '<?php $x = isset($this->foo) ? $this->foo : null;',
+            ],
+            [
+                '<?php $x = $this["foo"] ?? null;',
+                '<?php $x = isset($this["foo"]) ? $this["foo"] : null;',
+            ],
```

Comment:
But not `$thisIsSparta`, only `$this` usage (like provided examples) 😉.

Reply:
> It should check if it starts with $this rather than it's equal.

Why?? `$this->foo ?? null` works fine, only `$this ?? null` does not.

Reply:
`isset($this->foo) ? $this->foo : 42` and `$this->foo ?? 42` both trigger the error `Uncaught Error: Using $this when not in object context` so both code are equivalent, then PHP-CS-Fixer can suggest to replace one by another.

But it must not suggest it for exactly `isset($this)` and nothing else because in this exact case, `isset($this)` just return `false` while `$this ??` throw the error.

Reply:
Hmmm, I was pretty sure `isset()` would verify whole chain without error, but indeed it triggers an error for `isset($this->foo)` 🤔. My fault I did not check it before. It seems like an odd behavior, because both `isset($foo->bar->baz)` or `isset($foo['bar']['baz'])` check variable's existence and then go deeper. Why `$this` is different than that 🤷‍♂️?

But it can be "solved" with [`isset($this, $this->foo)`](https://3v4l.org/sQ681) and 😆. Interesting, I [tweeted about it](https://twitter.com/_Codito_/status/1668638444786970625) 😅 .

Anyway, multi-argument `isset()` won't be converted, so that's not a problem. I'll close the discussion, sorry for the noise.